packages/langium/src/services.ts

```diff
@@ -48,23 +47,36 @@ export type LangiumLspServices = {
     RenameHandler: RenameHandler
 }
 
-export type LangiumDefaultServices = {
-    parser: {
-        GrammarConfig: GrammarConfig
-        ValueConverter: ValueConverter
-        LangiumParser: LangiumParser
-        TokenBuilder: TokenBuilder
+export type LangiumGeneratedSharedServices = {
+    AstReflection: AstReflection
+}
+
+export type LangiumDefaultSharedServices = {
+    ServiceRegistry: ServiceRegistry
+    lsp: {
+        Connection?: Connection
     }
-    documents: {
+    workspace: {
         DocumentBuilder: DocumentBuilder
         LangiumDocuments: LangiumDocuments
         LangiumDocumentFactory: LangiumDocumentFactory
         TextDocuments: TextDocuments<TextDocument>
         TextDocumentFactory: TextDocumentFactory
+        IndexManager: IndexManager
+    }
+}
+
+export type LangiumSharedServices = LangiumDefaultSharedServices & LangiumGeneratedSharedServices
```

Comment:
Please add documentation: what are the shared services, how do they relate to the `LangiumServices`?